workers/reconciliationworker.py

```diff
@@ -0,0 +1,187 @@
+            ebsAccountNumber = entitlements.get_ebs_account_number(user.id)
+            logger.debug(
+                "Database returned %s account number for %s", str(ebsAccountNumber), user.username
+            )
+
+            # go to user api if no ebsAccountNumber is found
+            if ebsAccountNumber is None:
+                logger.debug("Looking up ebsAccountNumber for email %s", email)
+                ebsAccountNumber = user_api.lookup_customer_id(email)
+                logger.debug("Found %s number for %s", str(ebsAccountNumber), user.username)
+                if ebsAccountNumber:
+                    entitlements.save_ebs_account_number(user, ebsAccountNumber)
+
+            for sku in QuaySkuProperties:
+                # Check each sku since user can have multiple subscriptions
+                rh_sku = sku.name
+
+                if ebsAccountNumber is None:
+                    logger.debug("No account found for %s", user.username)
+                    break
+
+                subscription = marketplace_api.lookup_subscription(ebsAccountNumber, rh_sku)
+
+                # If there is no subscription, we need to create one for them
+                if subscription is None and user.stripe_id:
+                    logger.debug("Checking if we need to create subscription for %s", user.username)
+                    if self._check_stripe_matches_sku(user, sku):
+                        logger.debug("Will create %s for %s", rh_sku, user.username)
+                        marketplace_api.create_entitlement(ebsAccountNumber, rh_sku)
+                    continue
+                elif subscription is None and not user.stripe_id:
+                    logger.debug(
+                        "User %s is not on stripe, will not create entitlement for them",
+                        user.username,
+                    )
+                    continue
+
+                subscription_id = subscription["id"]
+                endDate = subscription["effectiveEndDate"]
+                # convert endDate from big int to datetime for adding to RedHatSubscriptions table
+                endDateDatetime = datetime.datetime.fromtimestamp(endDate / MILLISECONDS_IN_SECONDS)
+
+                # store subscription in the database if it is not already in there
+                local_subscription = entitlements.get_subscription_by_id(subscription_id)
+                if local_subscription is None:
+                    logger.debug(
+                        "Adding new subscription %s to database for user %s",
+                        rh_sku,
+                        user.username,
+                    )
+                    entitlements.save_subscription(
+                        user.id, subscription_id, ebsAccountNumber, endDateDatetime, rh_sku
+                    )
+                elif local_subscription.subscription_end_date < endDateDatetime:
+                    entitlements.update_subscription_end_date(subscription_id, endDateDatetime)
```

Comment:
Could we just keep track of the account number: what the purpose of persisting these subscriptions in the database?

Reply:
My understanding here is that we would look at the database every time we check stripe as well. That way we don't have to call the internal api too often.